go/libraries/doltcore/sqle/dolt_index_test.go

```diff
@@ -0,0 +1,637 @@
+		},
+		{
+			"twopk:idx_v2v1",
+			[]interface{}{3, 3},
+			[]interface{}{3, 4},
+			[]sql.Row{{1, 1, 3, 3}, {2, 2, 4, 3}},
+		},
+		{
+			"twopk:idx_v2v1",
+			[]interface{}{3, 4},
+			[]interface{}{4, 4},
+			[]sql.Row{{1, 2, 3, 4}, {2, 1, 4, 4}, {2, 2, 4, 3}},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("%s|%v%v", test.indexName, test.greaterThanOrEqual, test.lessThanOrEqual), func(t *testing.T) {
+			index, ok := indexMap[test.indexName]
+			require.True(t, ok)
+
+			expectedRows := convertSqlRowToInt64(test.expectedRows)
+
+			indexLookup, err := index.AscendRange(test.greaterThanOrEqual, test.lessThanOrEqual)
+			require.NoError(t, err)
+			dil, ok := indexLookup.(*doltIndexLookup)
+			require.True(t, ok)
+			indexIter, err := dil.RowIter(NewTestSQLCtx(context.Background()))
+			require.NoError(t, err)
+
+			var readRows []sql.Row
+			var nextRow sql.Row
+			for nextRow, err = indexIter.Next(); err == nil; nextRow, err = indexIter.Next() {
+				readRows = append(readRows, nextRow)
+			}
+			require.Equal(t, io.EOF, err)
+
+			assert.ElementsMatch(t, expectedRows, readRows)
+
+			indexLookup, err = index.DescendRange(test.lessThanOrEqual, test.greaterThanOrEqual)
+			require.NoError(t, err)
+			dil, ok = indexLookup.(*doltIndexLookup)
+			require.True(t, ok)
+			indexIter, err = dil.RowIter(NewTestSQLCtx(context.Background()))
+			require.NoError(t, err)
+
+			readRows = nil
+			for nextRow, err = indexIter.Next(); err == nil; nextRow, err = indexIter.Next() {
+				readRows = append(readRows, nextRow)
+			}
+			require.Equal(t, io.EOF, err)
+
+			assert.ElementsMatch(t, expectedRows, readRows)
+		})
+	}
+}
+
+func testDoltIndex(t *testing.T, keys []interface{}, expectedRows []sql.Row, compFunc func(keys ...interface{}) (sql.IndexLookup, error)) {
+	indexLookup, err := compFunc(keys...)
+	require.NoError(t, err)
+	dil, ok := indexLookup.(*doltIndexLookup)
+	require.True(t, ok)
+	indexIter, err := dil.RowIter(NewTestSQLCtx(context.Background()))
+	require.NoError(t, err)
+
+	var readRows []sql.Row
+	var nextRow sql.Row
+	for nextRow, err = indexIter.Next(); err == nil; nextRow, err = indexIter.Next() {
+		readRows = append(readRows, nextRow)
+	}
+	require.Equal(t, io.EOF, err)
+
+	assert.ElementsMatch(t, convertSqlRowToInt64(expectedRows), readRows)
+}
+
+func doltIndexSetup(t *testing.T) map[string]DoltIndex {
+	ctx := NewTestSQLCtx(context.Background())
+	dEnv := dtestutils.CreateTestEnv()
+	db := NewDatabase("dolt", dEnv.DoltDB, dEnv.RepoState, dEnv.RepoStateWriter())
+	root, err := dEnv.WorkingRoot(ctx)
+	if err != nil {
+		panic(err)
+	}
+	root, err = ExecuteSql(dEnv, root, `
+CREATE TABLE onepk (
+  pk1 BIGINT PRIMARY KEY,
+  v1 BIGINT,
+  v2 BIGINT
+);
+CREATE TABLE twopk (
+  pk1 BIGINT,
+  pk2 BIGINT,
+  v1 BIGINT,
+  v2 BIGINT,
+  PRIMARY KEY (pk1,pk2)
+);
+CREATE INDEX idx_v1 ON onepk(v1);
+CREATE INDEX idx_v2v1 ON twopk(v2, v1);
+INSERT INTO onepk VALUES (1, 1, 1), (2, 1, 2), (3, 3, 3), (4, 4, 3);
```

Comment:
This is a great start for these tests, but it should be expanded a bit:

1) Negative numbers
2) Other types (strings and dates would be good for a start).

1) should be part of this PR, but you could do a followup for 2).